api/uninstall/service.go

```diff
@@ -3,24 +3,34 @@ package uninstall
 import (
 	"context"
 	"fmt"
+	"time"
 
 	"github.com/gojekfarm/albatross/pkg/helmcli"
 	"github.com/gojekfarm/albatross/pkg/helmcli/flags"
 
 	"helm.sh/helm/v3/pkg/release"
 )
 
+const defaultTimeout = 300 * time.Second
+
 type Service struct {
 	cli helmcli.Client
 }
 
 // Uninstall a release according to the request provided and fails if req is incorrect.
 func (s Service) Uninstall(ctx context.Context, req Request) (Response, error) {
+	var timeout time.Duration
+	if req.Timeout < 1 {
```

Comment:
Can we check for 0 or nil depending on what the default value is?

Reply:
since the defaultValue is 0, we check for < 1 since negative values aren't supported either. Anything less than 1 defaults to 300 seconds. Since the application itself requires a default value of 300 seconds.

Reply:
IMO, allowing users to specify timeouts in minutes makes more sense here, because timeout of 2 seconds does not suffice for uninstall (it's very low for any helm action). Any integer > 0 specifying timeouts in minutes should be okay. What do you think? Otherwise we will need validations on permissible min value.

Reply:
The value was in seconds, since it's an API, the most probable use case for it would be programmatically, and we generally specify such time duration in seconds. The helm cli itself does not have any minimum permissible values, so I don't think we should implement a permissible min value either.